javaparser-core/src/main/java/com/github/javaparser/ast/expr/PatternExpr.java

```diff
@@ -21,8 +21,53 @@
 package com.github.javaparser.ast.expr;
 
 import com.github.javaparser.ast.AllFieldsConstructor;
+import java.util.Optional;
+import java.util.function.Consumer;
+import com.github.javaparser.ast.visitor.CloneVisitor;
+import com.github.javaparser.metamodel.PatternExprMetaModel;
+import com.github.javaparser.metamodel.JavaParserMetaModel;
+import com.github.javaparser.TokenRange;
 
 public abstract class PatternExpr extends Expression {
+
     @AllFieldsConstructor
-    public PatternExpr() { }
+    public PatternExpr() {
+    }
+
+    @Override
```

Comment:
Is it normal that there is no @Generated annotation? What will happen with subsequent generations? Isn't there a risk of inconsistency in the code?

Reply:
This is something I've noticed as well. Sometimes the `@Generated` annotation is added and sometimes not. It's also not entirely consistent across runs: I've seen `@Generated` added on a second or later codegen run, but I've never seen it removed. 

As far as I can tell, this annotation doesn't have any effect on the generators themselves, so the risk is that contributors will edit generated code without realising it. This is not an ideal situation, but I don't know what the solution is yet. I'm still learning how the generators work myself, so I'm hoping that adding the `RecordPatternExpr` class later will answer some of the questions I have as well.